feature_engine/timeseries/forecasting/expanding_window_features.py

```diff
@@ -170,12 +179,27 @@ def transform(self, X: pd.DataFrame) -> pd.DataFrame:
         # Common dataframe checks and setting up.
         X = super().transform(X)
 
-        tmp = (
-            X[self.variables_]
-            .expanding(min_periods=self.min_periods)
-            .agg(self.functions)
-            .shift(periods=self.periods, freq=self.freq)
-        )
+        # use pandas shift default autodetect
+        if self.fill_value is None:
```

Comment:
I would just pass fill_value to shift without the if clause. Pandas shift has their own if clauses to work with whatever the user passes to this param